src/client/datascience/raw-kernel/liveshare/hostRawNotebookProvider.ts

```diff
@@ -201,6 +207,14 @@ export class HostRawNotebookProvider
                 !kernelConnectionMetadata ||
                 (kernelConnectionMetadata?.kind === 'startUsingKernelSpec' && !kernelConnectionMetadata?.kernelSpec)
             ) {
+                sendTelemetryEvent(Telemetry.KernelSpecNotFoundError, undefined, {
+                    resourceType: getResourceType(resource),
+                    language: getTelemetrySafeLanguage(getLanguageInNotebookMetadata(notebookMetadata)),
+                    kernelConnectionProvided: !!kernelConnection,
+                    notebookMetadataProvided: !!notebookMetadata,
+                    hasKernelSpecInMetadata: !!notebookMetadata?.kernelspec,
+                    kernelConnectionFound: !!kernelConnectionMetadata
```

Comment:
Need to find out why these errors are being throw for webview notebooks.
Its possible with the recent fixes, the number of these cases would go down, but would like to ensure we have better data in place.